src/common/main.py

```diff
@@ -40,29 +41,38 @@ def main(self):
 
         self.output.start()
         self.file_input.start()
-        self.button_input.start()
+        self.button_input.run()
         self.mixing.start()
 
         #init UI
-        x = Thread(target=self.init_UI, args=(self.shutdown,))
+        x = Thread(target=self.init_UI, args=(self.shutdown,self.mixing))
         x.start()
 
-
         print("Type `quit` to quit")
 
-        while(True):
-            if (False and input() == 'quit') or not x.is_alive():  # Change false when developing in console mode
+        while(CONSOLE_MODE and x.is_alive()): 

```

Comment:
What happens when CONSOLE_MODE is false? Wouldn't the system shut down prematurely?

Reply:
tiny insignificant detail